### ISSUE TITLE
spurious errors on HTMLSourceElement interpretted as HTMLElement

### ISSUE THREAD

**Issue by BobFrGit** [Microsoft/TypeScript#4265]
`var source = <HTMLSourceElement> document.createElement("source");`
` source.src = path + fmt;`

Works fine but

`var source = document.createElement("source");`
`source.src = path + fmt;`
gets **error TS2339: Build: Property 'src' does not exist on type 'HTMLElement'.**

but if I hover over the "source" intellisense says **HTMLSourceElement**
![image](https://cloud.githubusercontent.com/assets/5490484/9185009/db2508a4-3f86-11e5-9de8-34495532e852.png)

This started happening after the latest update. I also notice posRight is gone.

**Comment by mhegazy** [comment]
which version of compiler are you using? `tsc --version`

**Comment by BobFrGit** [comment]
**Short answer 1.5.3**

If I go to dos (W10 command prompt) I get TS6029: Version 1.4.1.0

So I explicitly went to the visual studio prompt and got TS6029: Version 1.5.3

I just uninstalled VS203 but apparently it left debris so I uninstalled Typescript 1.4 explicitly.

Now I’m not getting the spurious error.

But when I go to DOS it still says “message TS6029: Version 1.4.1.0”??

Just as an FYI

Command Line
`where tsc.*`
`C:\Users\Bob\AppData\Roaming\npm\tsc`
`C:\Users\Bob\AppData\Roaming\npm\tsc.cmd``

VS prompt
`C:\Program Files (x86)\Microsoft Visual Studio 14.0>where tsc.*`
`C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.5\tsc.exe`
`C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.5\tsc.js`
`C:\Users\Bob\AppData\Roaming\npm\tsc`
`C:\Users\Bob\AppData\Roaming\npm\tsc.cmd`

**Comment by mhegazy** [comment]
Locally, i get this error message only on 1.4. 1.5* seems to have it fixed.

here is the code i am trying:

```ts
var source = document.createElement("source");
source.src = "s";
```

**Comment by BobFrGit** [comment]
I wonder why VS2015 was finding the old version not the new one that was in its search path.

**Comment by mhegazy** [comment]
do you happen, by any chance, `HKEY_CURRENT_USER\SOFTWARE\Microsoft\VisualStudio\14.0\TypeScriptLanguageService\Custom*` in your registry?

**Comment by BobFrGit** [comment]
No such entry. Of course it could’ve been removed by a uninstall.

**Comment by BobFrGit** [comment]
I replied to the previous question in email but don't see it here -- don't have such a registerly etnry.

But as an FYI I found this line in a different csprroj file

`    <Exec Command="&quot;$(PROGRAMFILES)\Microsoft SDKs\TypeScript\1.4\tsc&quot; --sourcemap  --target ES5  @(TypeScriptCompile ->'&quot;%(fullpath)&quot;' , ' ')" Condition="'@(TypeScriptCompile)'!=''" />`

guess updating is not perfect. Manually changed the 4 to a 5

**Comment by mhegazy** [comment]
So this is not an issue any longer?

**Comment by BobFrGit** [comment]
I seem to be fine now.

**Comment by mhegazy** [comment]
thanks!
